src/main/java/com/gradle/CommonCustomUserDataGradlePlugin.java

```diff
@@ -39,7 +39,8 @@ private void applySettingsPlugin(Settings settings) {
         settings.getPluginManager().withPlugin("com.gradle.enterprise", __ -> {
             CustomGradleEnterpriseConfig customGradleEnterpriseConfig = new CustomGradleEnterpriseConfig();
 
-            GradleEnterpriseExtension gradleEnterprise = settings.getExtensions().getByType(GradleEnterpriseExtension.class);
+            Object extension = settings.getExtensions().getByName("gradleEnterprise");
+            GradleEnterpriseExtension gradleEnterprise = ProxyFactory.createProxy(extension, GradleEnterpriseExtension.class);
```

Comment:
❔ could we check if the classloaders are compatible and only fall back to reflection if they are not?

Reply:
I discussed this with Luke and we agreed to always do this in order to get more coverage and spot potential issues sooner.